Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs

```diff
@@ -95,19 +97,80 @@ public override void Execute(CancellationToken cancellationToken)
             }
         }
 
+        private static volatile FileIntegrity[] files;
+
         private void ExecuteInternal(CancellationToken cancellationToken)
         {
             DebugLogger.Log("Checking version integrity.");
 
             _status.IsActive.Value = true;
 
-            var files = new FileIntegrity[_versionSummary.Files.Length];
-
-            for (int i = 0; i < _versionSummary.Files.Length; i++)
+            files = new FileIntegrity[_versionSummary.Files.Length];
+            int i = 0;
+            foreach (var file in _versionSummary.Files)
             {
-                files[i] = CheckFile(_versionSummary.Files[i]);
+                while (true)
+                {
+                    int tmp;
+                    lock (this)
+                    {
+                        tmp = _space;
+                    }
+
+                    if (tmp < 16)
```

Comment:
Please introduce a constant `MaxThreadCount` instead of `16` here.

Reply:
Also, please consider semaphores - https://docs.microsoft.com/en-us/dotnet/api/system.threading.semaphore?view=net-5.0